concreate/descriptor.py

```diff
@@ -116,13 +120,18 @@ def _process_ports(self):
         for port in self.descriptor['ports']:
             port['name'] = port['value']
 
-    def _process_dependencies(self):
-        """ Generate name attribute for dependencies """
-        dependencies = {}
-        for dependency in self.descriptor['dependencies']:
-            resource = Resource.new(dependency, self.directory)
-            dependencies[resource.name] = resource
-        self.descriptor['dependencies'] = dependencies
+    def _process_module_repositories(self):
```

Comment:
I really think it's better to just define a local relative repository, e.g.:
```yaml
modules:
  repositories:
  - name: java
    path: modules
```
This is explicit and doesn't require any extra magic.  It also allows them to be overridden, if that is desirable, e.g.
```yaml
modules:
  repositories:
  - name: java.platform
    path: modules.rhel
```
where ``modules.rhel`` might look like
```yaml
modules:
  install:
  - module: java.common
```
and could be overridden:
```yaml
modules:
  repositories:
  - name: java.platform
    module: java.centos
```
Which brings up a question, are we using ``name`` for the installed modules, because I think I like what I wrote above (using ``module``), as it's self describing.